src/nets/e2e_asr_attctc_th.py

```diff
@@ -697,6 +697,27 @@ def recognize(self, h, recog_args):
 
         return y_seq
 
+    # end detection desribed in Eq. (50) of
+    # S. Watanabe et al "Hybrid CTC/Attention Architecture for End-to-End Speech Recognition"
+    def end_detect(self, ended_hyps, i, M=3, D_end=np.log(1 * np.exp(-10))):
```

Comment:
I think this function should be in somewhere outside e2e_asr_attctc[_th].py and be shared from them like `from e2e_common import end_detection`. Because this function seems to be static (not use self) and be free from numpy/torch operations.